### ISSUE TITLE
bd CPS: refactor bd_CPS_grapher

### ISSUE THREAD

**Issue by bdecon** [bdecon/econ_data#63]
The bd_CPS_grapher has the combination of lots of code and not that many features. With a few hours of work, the code can be reduced and the features expanded. This issue is to reduce the code and clean it up a bit, with more comments and less repetition.

If possible, also speed up the production of charts, possibly by reducing the the number of columns read by pandas or the number of times each annual file in read into memory.

**Comment by bdecon** [comment]
I've spent half a day working through the file. The end result is possibly even more LOC, and I haven't done much with the graph settings, but for most calculations the result is ~1.5 seconds compared to 6 before. I fixed an error affecting weekly wage estimates (previously they were only calculated for persons with hourly wages > 0). I also fixed the grapher to run with pyarrow 0.12 and pandas 0.24. 

So not yet closing the issue, but have made a ton of progress.
